### ISSUE TITLE
Bug “Add or edit Image”

### ISSUE THREAD

**Issue by BulletQQ** [stephanrauh/ngx-extended-pdf-viewer#2138]
Hi @stephanrauh , there is a small but unpleasant error, if you go to your demo site ([demo site](https://pdfviewer.net/extended-pdf-viewer/simple)), click on the “Add or edit Image” button, after clicking on it, a file upload window will immediately appear (this is not correct), the file upload window should appear after clicking on the “Add” button Image"

**Comment by stephanrauh** [comment]
Hm. At the moment, the behavior of the application isn't that surprising, because there's only a single menu entry. In other words, at the moment, the menu is superfluous. But you're right: I suppose there will be more menu entries in the future, so I ought to fix this. Although I'd prefer to get rid of the useless menu, but I'm afraid it might backfire in a few versions.

**Comment by stephanrauh** [comment]
I'm trying to fix this, but this particular bug is nasty. Something strange is going on here. When I deactivate the upload window on the toolbar button, it won't work in the popup menu, either. I'm puzzled!

**Comment by BulletQQ** [comment]
Hi @stephanrauh , I noticed that in viewer.js #bindListeners(buttons) dispatch is passed evt.detail = 0 which in turn makes isFromKeyboard = true, but if you try to make isFromKeyboard "false" then everything will work as it should

**Comment by stephanrauh** [comment]
Whow. I believe you're up to something. At the moment, I don't know how to pass the `detail` parameter, and I'd also like to research what it means before tinkering around with it. So I need some time to implement the solution. Nonetheless, it seems you've found the solution of the issue!

**Comment by BulletQQ** [comment]
@stephanrauh To fix this problem, you need to remove this line from PdfStampEditorComponent `document.getElementById('editorStamp')?.click();`
and replace with this

```
var element = document.getElementById('editorStamp');

var clickEvent = new MouseEvent('click', {
     'view' : window,
     'bubbles': true,
     'cancelable': true,
     'detail':1
   });

element.dispatchEvent(clickEvent);
```

The problem is that you artificially call the mouse click, since when the mouse is physically clicked in the PointerEvent, the exact number of clicks is recorded in the detail fields

**Comment by stephanrauh** [comment]
Awesome. I saw the `detail` bit, too, but I didn't know what it means. Before I managed to find out, you've come up with a solution. That's the spirit. Carry on like this!

Your bugfix has landed with version 19.2.2.

Best regards
Stephan
